Component/tertiary button

merryface opened the pull request:
#8 
1. Created button with correct styling

2. copied example component for props:
-   type = submit/button (defaulted to button)
-   onClick
-   content

3. added prop of "colour". defaulted to 'blue.' any other value will turn it to the alternate colour

merryface committed:
component generated

merryface committed:
button styled

merryface committed:
conditional border colour added.

merryface committed:
changed p to button with submit type

merryface committed:
made component a button that takes in a function as prop

merryface committed:
git merge develop interrupted

RobTF9 commented on src/components/TertiaryButton/TertiaryButton.jsx:
```diff
@@ -0,0 +1,23 @@
+
+const TertiaryButton = ({ content, colour, onClick, type = 'button' }) => {
+  return (
+    <TertiaryButtonWrapper colour={colour}>
+      <button type={type === 'submit' ? 'submit' : 'button'} onClick={onClick}>
+        {content}
+      </button>
+    </TertiaryButtonWrapper>
+  );
+};
```
Change wrapper to button component.

RobTF9 commented on src/components/TertiaryButton/TertiaryButton.styles.js:
```diff
@@ -0,0 +1,17 @@
+  padding: var(--smallest-space);
+  text-align: center;
+
+  border: 1px solid;
```
1px > 0.1rem

RobTF9 commented on src/components/TertiaryButton/TertiaryButton.styles.js:
```diff
@@ -0,0 +1,17 @@
+import styled from 'styled-components';
+
+export const TertiaryButtonWrapper = styled.div`
+  padding: var(--smallest-space);
```
Double check paddding matches designs.

RobTF9 commented on src/components/TertiaryButton/TertiaryButton.styles.js:
```diff
@@ -0,0 +1,17 @@
+  border-color: ${(prop) =>
+    prop.colour === 'blue' ? 'var(--lightblue)' : 'var(--logopink)'};
+  box-sizing: border-box;
+  border-radius: 100px;
```
100px > 10rem

RobTF9 commented on src/components/TertiaryButton/TertiaryButton.styles.js:
```diff
@@ -0,0 +1,17 @@
+  border: 1px solid;
+  border-color: ${(prop) =>
+    prop.colour === 'blue' ? 'var(--lightblue)' : 'var(--logopink)'};
+  box-sizing: border-box;
```
Remove box-sizing, applied by default

RobTF9 commented on src/components/TertiaryButton/TertiaryButton.test.jsx:
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import TertiaryButton from './TertiaryButton';
+
+describe('TertiaryButton', () => {
+  test('renders TertiaryButton component', () => {
+    render(<TertiaryButton value="Tertiary Button" />);
+
+    expect(screen.getByText('Tertiary Button'));
+  });
+});
```
Test click handler

merryface committed:
resolved most review items. now testing

merryface committed:
tests written

merryface committed:
additional styling corrections

merryface committed:
changed border to 0.2 on focus and active

merryface commented:
fixed all issues as pointed out by first review. Tests were copied from Primary Button

1. Added additional styling to be more consistent with primary button while respecting the figma design
2. have no idea how to run tests (never used jest or anything like that before)

merryface committed:
added imports to test

shanesc commented:
> fixed all issues as pointed out by first review. Tests were copied from Primary Button
> 
> 1. Added additional styling to be more consistent with primary button while respecting the figma design
> 2. have no idea how to run tests (never used jest or anything like that before)
> 
> was unsure if I should use medium or large lateral padding. Picked the closest looking one (medium)

Padding looks good.
FYI if you hover of a component in Figma you can see distances from other elements:
![image](https://user-images.githubusercontent.com/47223963/106303515-20ad4a00-6252-11eb-9420-c8ca28746ba5.png)

shanesc commented on src/components/TertiaryButton/TertiaryButton.stories.jsx:
```diff
@@ -0,0 +1,11 @@
+};
+
+export const tertiaryButtonWithContent = () => (
+  <TertiaryButton content="Tertiary Button" colour="not blue" />
```
Remove `colour="not blue"`, not needed

shanesc commented on src/components/TertiaryButton/TertiaryButton.styles.js:
```diff
@@ -0,0 +1,22 @@
+
+export const TertiaryButtonWrapper = styled.button`
+  padding: var(--smallest-space) var(--medium-space);
+
+  text-align: center;
+
```
Delete the empty line break whitespace to be consistent with other CSS

shanesc commented on src/components/TertiaryButton/TertiaryButton.styles.js:
```diff
@@ -0,0 +1,22 @@
+
+  text-align: center;
+
+  border: 0.1rem solid;
+  border-color: var(--lightblue);
```
remove the `border-color` property and move `var(--light-blue)` up to the `border` property
i.e.
`border: 0.1rem solid var(--lightblue);`

shanesc commented on src/components/TertiaryButton/TertiaryButton.jsx:
```diff
@@ -0,0 +1,19 @@
+TertiaryButton.propTypes = {
+  content: PropTypes.string.isRequired,
+  type: PropTypes.oneOf(['submit', 'button', 'reset']),
+  onClick: PropTypes.func
```
I asked @ammersive about this as well, what do you both think about having the `onClick` handler prop being required on buttons? That way there's less likelihood of a bug with a button not performing an action.

Happy with it either way we decide, just a thought.

shanesc replied:
Scratch that, not needed. Here was Rob's response:

> No, you don't necessarily need one. In forms for example, the event is handled my the submission of the form as opposed to the button so a onClick is not always required.

merryface replied:
frankly i literally copied your tests for the primary button, I'm still studying them in my own time and not entirely comfortable as to how and when to do them, So I'm glad Rob said no need :P

shanesc commented on src/components/TertiaryButton/TertiaryButton.test.jsx:
```diff
@@ -0,0 +1,29 @@
+
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+});
```
Tests look good!

shanesc reviewed (changes requested):
Overall looks really good, just a couple of style changes to be made.

merryface committed:
refactor: completed Shane's change items

shanesc reviewed (approved):
Looks good, co ahead and click "Squash and Merge" 👍 

*edit: just saw that you can't merge until @RobTF9 and I both approve the reviews*

RobTF9 reviewed (commented):
Looks good to me 👍

merryface commented:
Rob is the workflow to delete the branch afterwards?

RobTF9 commented:
Yep
